compiler/passes/convert-typed-uast.cpp

```diff
@@ -0,0 +1,2041 @@
+   the callDestructors pass.
+ */
+
+#include "convert-uast.h"
+
+#include "CForLoop.h"
+#include "CatchStmt.h"
+#include "DecoratedClassType.h"
+#include "DeferStmt.h"
+#include "DoWhileStmt.h"
+#include "ForLoop.h"
+#include "ForallStmt.h"
+#include "IfExpr.h"
+#include "ImportStmt.h"
+#include "LoopExpr.h"
+#include "ParamForLoop.h"
+#include "ResolveScope.h"
+#include "TemporaryConversionThunk.h"
+#include "TryStmt.h"
+#include "WhileDoStmt.h"
+#include "build.h"
+#include "config.h"
+#include "global-ast-vecs.h"
+#include "metadata.h"
+#include "optimizations.h"
+#include "parser.h"
+//#include "passes.h"
+#include "resolution.h"
+#include "stmt.h"
+
+
+#include "view.h" // DEBUG only
+
+#include "chpl/framework/compiler-configuration.h"
+#include "chpl/framework/global-strings.h"
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/resolution/ResolvedVisitor.h"
+#include "chpl/resolution/resolution-queries.h"
+#include "chpl/types/all-types.h"
+#include "chpl/uast/all-uast.h"
+#include "chpl/uast/chpl-syntax-printer.h"
+#include "chpl/util/assertions.h"
+#include "chpl/util/string-escapes.h"
+
+#include "convert-help.h"
+
+#include "llvm/ADT/SmallPtrSet.h"
+
+#include <iostream>
+
+using namespace chpl;
+using namespace resolution;
+using namespace uast;
+
+// converts resolved uAST + types to typed AST
+struct TConverter final : UastConverter {
+
+  /// Nested Types ///
+  using RV = ResolvedVisitor<TConverter>;
+
+  // State required to mimic the desugaring of multi-declarations into
+  // regular ones.
+  struct MultiDeclState {
+    // The result of computing the target locale, stored in a temp, used
+    // for remote variables.
+    Symbol* localeTemp = nullptr;
+
+    Symbol* typeTemp = nullptr;
+    Symbol* prev = nullptr;
+    Expr* prevTypeExpr = nullptr;
+    Expr* prevInitExpr = nullptr;
+
+    void reset() {
+      typeTemp = nullptr;
+      prev = nullptr;
+      prevTypeExpr = nullptr;
+      prevInitExpr = nullptr;
+    }
+
+    // For remote variables, these helpers don't modify the def point, since
+    // def point is quite different (it's an invocation of a remote variable
+    // wrapper builder with extra arguments).
+
+    void replaceTypeExpr(Expr* newExpr) {
+      if (localeTemp) {
+        prevTypeExpr->replace(newExpr);
+      } else {
+        prev->defPoint->exprType = newExpr;
+      }
+      prevTypeExpr = nullptr;
+    }
+
+    void replaceInitExpr(Expr* newExpr) {
+      if (localeTemp) {
+        prevInitExpr->replace(newExpr);
+      } else {
+        prev->defPoint->init = newExpr;
+      }
+    }
+  };
+
+
+  /// Fields ///
+  Context* context = nullptr;
+
+  // What is the module / function we are converting?
+  const AstNode* symbol = nullptr; // Module* or Function*
+  FnSymbol* curFnSymbol = nullptr; // corresponding to the above
+  ModuleSymbol* curModuleSymbol = nullptr; // corresponding to the above
+  Symbol* rvv = nullptr;
```

Comment:
`rvv` is the return value temp, right? I think moving away from the name `rvv` would be helpful here.

Reply:
`rvv` is specific and used in the production compiler. Most concretely, the return value variable is the variable with `FLAG_RVV`.

I will add a comment to say that it's the return value variable.

Reply:
On second thought, I'll rename it to `curRetVar`, since it is a field, and `rvv` might be too obtuse when looking at methods here.